### PR TITLE
feat: add ability to select locale

### PR DESCRIPTION
**Pull request by azat-io**
Hello!

I am the author of [ESLint Plugin Perfectionist](https://github.com/azat-io/eslint-plugin-perfectionist). I am currently thinking of using your package inside my ESLint plugin to provide natural sorting.

However, there are a few things in your plugin that I am missing. Could you please look into my poolquest?

### CODE REVIEW

**Review by yobacca — COMMENTED** [review]
Why have you chose to add the locale to the `compare` callback function and not extending the `CompareOptions`. Do you need to be able to compare values with  different locales at the same time?

Additiionally I'd like to add this feature to the `orderBy` function as well to have a feature parity here.

File: src/utils/compareChunks.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@ import type { ChunkMaps } from '../types';
 export const compareChunks = (
   chunksA: ChunkMaps,
   chunksB: ChunkMaps,
+  locale = 'en',
```
[the remark this answers is not in the document]
Comment (azat-io) [reply src/utils/compareChunks.ts]: Fixed

File: src/utils/compareUnicode.ts [hunk 2]
```diff
@@ -1,4 +1,8 @@
-export const compareUnicode = (stringA: string, stringB: string): number => {
-  const result = stringA.localeCompare(stringB);
+export const compareUnicode = (
+  stringA: string,
+  stringB: string,
+  locale = 'en',
```
[the remark this answers is not in the document]
Comment (azat-io) [reply src/utils/compareUnicode.ts]: Fixed

File: src/utils/compareValues.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ import type { MappedValueRecord } from '../types';
 export const compareValues = (
   valueA: MappedValueRecord,
   valueB: MappedValueRecord,
+  locale = 'en',
```
[the remark this answers is not in the document]
Comment (azat-io) [reply src/utils/compareValues.ts:9]: Fixed

**Comment by azat-io** [comment]
@yobacca, Looks like I move locale to `CompareOptions`.

**Comment by azat-io** [comment]
@yobacca We are planning to do a major release of [Perfectionist](https://github.com/azat-io/eslint-plugin-perfectionist) within the next two days (switching to your library is a breaking change for us). Please check the pull request and release new version please. 

Then we will be able to take this package to dependencies in that release.

**Comment by yobacca** [comment]
Yes, I will provide a new release by today.

**Comment by yobacca** [comment]
New release is now available.

**Pull request closed without merging**